crates/transaction-pool/src/maintain/mod.rs

```diff
@@ -0,0 +1,363 @@
+        if drift_monitor.state().is_drifted() {
+            // assuming all senders are dirty
+            drift_monitor.set_dirty_addresses(pool.unique_senders());
+            // make sure we toggle the state back to in sync
+            drift_monitor.set_state(PoolDriftState::InSync);
+        }
+
+        // if we have accounts that are out of sync with the pool, we reload them in chunks
+        if drift_monitor.has_dirty_addresses() && !drift_monitor.is_reloading() {
+            drift_monitor.start_reload_accounts(
+                client.clone(),
+                pool_info.last_seen_block_hash,
+                &task_spawner,
+            );
+        }
+
+        // check if we have a new finalized block
+        if let Some(BlobStoreUpdates::Finalized(blobs)) = canon_processor.update_finalized(&client)
+        {
+            metrics.inc_deleted_tracked_blobs(blobs.len());
+            // remove all finalized blobs from the blob store
+            pool.delete_blobs(blobs);
+            // and also do periodic cleanup
+            let pool = pool.clone();
+            task_spawner.spawn_blocking(Box::pin(async move {
+                debug!(target: "txpool", "cleaning up blob store");
+                pool.cleanup_blobs();
+            }));
+        }
+
+        // select of account reloads and new canonical state updates which should arrive at the rate
+        // of the block time
+        tokio::select! {
+            // handle reloaded accounts
+            _ = async { futures_util::future::pending::<()>().await }, if !drift_monitor.is_reloading() => {
+                // this branch will never execute, it's just a placeholder
+                // for when we don't have account reloads running
+            }
+            _ = async {}, if drift_monitor.is_reloading() => {
+                if let drift_monitor::DriftMonitorResult::AccountsLoaded(LoadedAccounts { accounts, .. }) = drift_monitor.process_reload_result() {
+                     // update the pool with the loaded accounts
+                     pool.update_accounts(accounts);
+                 }
+            }
+
+            // handle new canonical events
+            ev = events.next() => {
+                if ev.is_none() {
+                    // the stream ended, we are done
+                    break;
+                }
+
+                // on receiving the first event on start up, mark the pool as drifted to explicitly
+                // trigger revalidation and clear out outdated txs.
+                drift_monitor.on_first_event();
+
+                // process the event
+                if let Some(event) = ev {
+                    // try processing as a reorg first
+                    if !canon_processor.process_reorg(event.clone(), &client, &pool, &mut drift_monitor).await {
+                        // if not a reorg, try as a commit
+                        canon_processor.process_commit(event, &client, &pool, &mut drift_monitor);
+                    }
```

Comment:
this if else check feels a bit strange, could we convert this into something like

`processor.on_event` which does the match internally?